lib/assertions.js

```diff
@@ -120,6 +120,12 @@ module.exports = function (expect) {
         expect(subject instanceof Constructor, '[not] to be truthy');
     });
 
+    expect.addAssertion('<any> [not] to be one of <any>', function (expect, subject, superset) {
+        if ((superset.indexOf(subject) !== -1) === expect.flags.not) {
```

Comment:
`Array.prototype.indexOf` uses === sematics, which means that these fail:

```js
expect(NaN, 'to be one of', [NaN]);
UnexpectedError: 
expected NaN to be one of [ NaN ]
```

```
expect(-0, 'not to be one of', [0]);
UnexpectedError: 
expected -0 not to be one of [ 0 ]
```

Could you switch to `Object.is` instead like [`to be`](http://unexpected.js.org/assertions/any/to-be/)? We have a polyfill that's already imported as `objectIs` at the top of assertions.js, so it's probably something like (untested):

```js
expect(superset.some(function (item) {
    return objectIs(subject, item);
}), '[not] to be true');
```